code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/TokenUtils.java

```diff
@@ -95,6 +104,17 @@ public Account getAccountFromJWT() {
         return jsonData;
     }
 
+    private String retrieveJwt(String jwtKey) {
+        AuthToken authToken= authTokenDao.getTokenById(jwtKey);
+        if (authToken == null) {
+            return null;
+        }
+        if (!authToken.getProvider().equalsIgnoreCase(SecurityConstants.AUTH_PROVIDER.get())){
+            throw new ClientVisibleException(ResponseCodes.UNAUTHORIZED);
```

Comment:
That would ultimately throw a 401 still. This needs to be a 401 because the authProvider has changed and it needs to be explicitly stored.